onnxruntime/core/providers/qnn/builder/opbuilder/base_op_builder.cc

```diff
@@ -60,7 +59,7 @@ Status BaseOpBuilder::ProcessInput(QnnModelWrapper& qnn_model_wrapper,
   }
 
   OnnxInputInfo input_info = {};
-  ORT_RETURN_IF_ERROR(qnn_model_wrapper.GetOnnxInputInfo(input, is_quantized_model, input_info));
+  ORT_RETURN_IF_ERROR(qnn_model_wrapper.GetOnnxInputInfo(input, is_quantized_node, input_info));
```

Comment:
I think we can get rid of the `is_quantized_node` parameter from `ProcessInput` and `qnn_model_wrapper.GetOnnxInputInfo` if we inspect the individual input (which is of type `NodeUnitIODef`). For example, inside `GetOnnxInputInfo`:

```C++
Status QnnModelWrapper::GetOnnxInputInfo(const NodeUnitIODef& input, OnnxInputInfo& input_info) const {
  const std::string& name = input.node_arg.Name();
  const bool is_quantized_input = input.quant_param.has_value();  // !!! Can inspect to determine if the input is quantized.

  // Fill in quantization param info.
  input_info.quant_param = QNN_QUANTIZE_PARAMS_INIT;
  utils::InitializeQuantizeParam(input_info.quant_param, is_quantized_input);

  if (is_quantized_input) {
    ORT_RETURN_IF_NOT(ProcessQuantizationParameter(input.quant_param,
                                                   input_info.quant_param.scaleOffsetEncoding.scale,
                                                   input_info.quant_param.scaleOffsetEncoding.offset),
                      "QNN EP: Cannot get quantization parameters for input ", name.c_str());
  }

  // Fill in QNN data type ...
}
```

Reply:
good idea.

---
In reply to: [1303647213](https://github.com/microsoft/onnxruntime/pull/17194#discussion_r1303647213) [](ancestors = 1303647213)